kong/db/schema/init.lua

```diff
@@ -655,8 +655,57 @@ Schema.entity_checkers = {
     fn = function(entity, arg)
       return arg.fn(entity)
     end,
-  }
+  },
+
+  mutually_required = {
+    run_with_missing_fields = true,
+    fn = function(entity, field_names)
+      local nonempty = {}
+
+      for _, name in ipairs(field_names) do
+        if is_nonempty(get_field(entity, name)) then
+          insert(nonempty, name)
+        end
+      end
+
+      if #nonempty == 0 or #nonempty == #field_names then
+        return true
+      end
 
+      return nil, "All or none of " .. quoted_list(field_names) .. " must be set. "
+                  .. "Only " .. quoted_list(nonempty) .. " found"
```

Comment:
Two very small things here:
* the error string needs to start with lowercase (`all or`...)
* please add the error string to `validation_errors` use `string.format` on that (it keeps the schema error messages consistent, and who knows, someday we might even internationalize them!)

Reply:
Done. I slightly changed the errors messages to make them consistent with other entity checkers.